### PR TITLE
etcd-tester: refactor

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5254.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @xiang90 @heyitsanthony 

Thanks.

**Comment by gyuho** [comment]
The functional-test needs clean-up first. WIP.

**Comment by gyuho** [comment]
/cc @xiang90 PTAL.

Just refactored with fix for the issue.

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
...
 		tt.status.Failures = append(tt.status.Failures, f.Desc())
 	}
-	for i := 0; i < tt.limit; i++ {
-		tt.status.setRound(i)
+	round := 0
+	for {
+		tt.status.setRound(round)
+		tt.status.setCase(-1)
 		roundTotalCounter.Inc()
+		lPrefix := tt.logPrefix()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: remove this var? just use tt.logPrefix()? i think it is more obvious.

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
...
 
-		var (
-			currentRevision int64
-			failed          bool
-		)
+		var failed bool
 		for j, f := range tt.failures {
 			caseTotalCounter.WithLabelValues(f.Desc()).Inc()
 			tt.status.setCase(j)
+			prefix := tt.logPrefix()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: same here.

File: tools/functional-tester/etcd-tester/tester.go [hunk 3]
```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
...
+		time.Sleep(time.Second)
+	}
+	plog.Printf("%s updated current revisions...", prefix)
 }
 
-func (s *Status) setCase(c int) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	s.Case = c
+func (tt *tester) doConsistencyCheck() (failed bool) {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: checkConsistency

File: tools/functional-tester/etcd-tester/tester.go [hunk 4]
```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
...
-			if !ok || rerr != nil {
-				plog.Printf("[round#%d case#%d] checking current revisions failed [revisions: %v]", i, j, revs)
-				if err := tt.cleanup(i, j); err != nil {
-					plog.Printf("[round#%d case#%d] cleanup error: %v", i, j, err)
-					return
-				}
-				failed = true
-				break
+				tt.startStressers()
+			} else if j == len(tt.failures)-1 {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: i feel we should just remove the else if and always update current revision. it is no harm anyway.

File: tools/functional-tester/etcd-tester/tester.go [hunk 5]
```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
...
-
-			plog.Printf("[round#%d case#%d] restarting the stressers...", i, j)
-			for _, s := range tt.cluster.Stressers {
-				go s.Stress()
-			}
-
-			plog.Printf("[round#%d case#%d] succeed!", i, j)
+			plog.Printf("%s succeed!", prefix)
 		}
+		tt.status.setCase(-1)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: why set case to -1?

**Comment by gyuho** [comment]
Redeployed with this branch and confirmed that https://github.com/coreos/etcd/issues/5254 is fixed.

Redeploying testing cluster after this merge now.

**Pull request merged**